test/cypress/integration/tests/scenario-exec-spec.js

```diff
@@ -273,6 +369,137 @@ describe('Scenario Execution', function () {
     validateNetCharEvent(elemType, name, l, lv, pl, tp, tp-1, dist);
   }
 
+  // Save Replay Events
+  function testSaveReplayEvents(name, desc) {
+    cy.log('Save events');
+    click(meep.EXEC_BTN_EVENT);
+    click(meep.EXEC_BTN_EVENT_BTN_SAVE_REPLAY);
+    type(meep.MEEP_DLG_SAVE_REPLAY_NAME, name);
+    type(meep.MEEP_DLG_SAVE_REPLAY_DESCRIPTION, desc);
+    click(meep.MEEP_DLG_SAVE_REPLAY, 'Ok');
+    cy.wait(1000);
+    //no way to check except by executing auto-replay test that gets the file
+    click(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY);
+    select(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_EVT_REPLAY_FILES, name);
+    cy.wait(500);
+    //check that by loading the saved replay event file that the description gets populated to confirm it was stored properly
+    //and check play button is enabled
+    cy.contains(desc)
+    verifyEnabled(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_BTN_REPLAY_START, true);
+  }
+
+  // Auto Replay
+  function testAutoReplay(name) {
+    cy.log('Auto Replay');
+
+    testAutoReplayExecution(name, false)
+    cy.wait(1000);
+    //confirm execution started
+    verifyEnabled(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_BTN_REPLAY_START, false);
+    cy.wait(75000);
+    //confirm it executed completely
+    verifyEnabled(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_BTN_REPLAY_START, true);
+
+    testAutoReplayExecution(name, true)
+    cy.wait(1000);
+    //confirm execution started
+    verifyEnabled(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_BTN_REPLAY_START, false);
+    cy.wait(75000);
+    //confirm it executed completely but restarted
+    verifyEnabled(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_BTN_REPLAY_START, false);
+
+    click(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_BTN_REPLAY_STOP)
+    cy.wait(500);
+    verifyEnabled(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_BTN_REPLAY_START, true);
+  }
+
+  function testAutoReplayExecution(name, loop) {
+    click(meep.EXEC_BTN_EVENT);
+    click(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY);
+    if (loop) {
+      check(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_CHKBOX_LOOP, true);
+    } else {
+      check(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_CHKBOX_LOOP, false);
+    }
+    select(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_EVT_REPLAY_FILES, name);
+    cy.wait(500);
+    verifyEnabled(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_BTN_REPLAY_START, true);
+    click(meep.EXEC_BTN_EVENT_BTN_AUTO_REPLAY_BTN_REPLAY_START)
+    //check right number of events will be executed
+    cy.contains("EVENT COUNT");
+    cy.contains("/ 13");
```

Comment:
It is difficult to maintain this. If we update the mobility events then this number must be changed.

To fix this we would need to restart the scenario and send a smaller number of events (e.g. 1 mobility + 1 net char). We would then always have these 2 events only. However, the scenario teardown & bring-up is what takes the most time, which would make the overall test longer.

Recommendation: leave as is for now, but add comments in code to clearly state where this event count is coming from and how to calculate it.

Reply:
Also, I believe the 'verify()' call can be used to search for text contained within a specific element. We should try to verify that only the status bar contains this text.

Reply:
fixed... added comments and also number of events now passed in param to the function.. so it is clear.. also added a label to check rather than whole page